Entities/Scene.cpp

```diff
@@ -2891,7 +2905,25 @@ void Scene::ResetPathFinding()
 
 void Scene::UpdatePathFinding()
```

Comment:
Assuming it's not too unwieldy, I'd rather have the method comment explain what's going on in a bit more detail, rather than all these inline comments.

I.e. explain that the method updates the shared pathfinder and, if there are changes, updates the individual ones, ensuring that doors for each team are ignored.

Reply:
In this case I'd prefer that they're inline. Externally, the only thing that a user of the function needs to know is that this recalculates outdated pathing data - how it does that in this case (using several pathfinders and swapping around door materials and all that crap) are entirely internal implementation details that could change.

Furthermore, having the comments be next to the actual lines of code means that if or when things do change, it's easier to remember to update them.

Reply:
Hmmm, I don't really agree. Inline comments are valuable to explain
1. Why a seemingly odd decision was made
2. What some piece of unclear, complicated (or math-heavy) code is doing

If you need them for stuff other than that, things need to be renamed and fiddled with to make the code clearer, imo.

For the most part, these comments don't really do either of those 2 things. They're not as useless as some dancomments that would just, for example, say something like:
```cpp
// Override door material for team
g_MovableMan.OverrideMaterialDoors(true, team);
````
But they're also mostly explaining pretty clear things.
I guess the only part of it that's at all non-obvious is the shared pathfinder being NoTeam, and I'd rather that be more of a note that the shared pathfinder uses NoTeam, instead of basically saying "`RecalculateAreaCosts` updates the pathfinder". Also, I think the shared pathfinder using NoTeam is explained in other places anyway, right?

Reply:
I think that the overriding material doors being used for team-specific pathing is not obvious. I didn't understand it initially, so that's why I'm being explicit.

> Also, I think the shared pathfinder using NoTeam is explained in other places anyway, right?

The concept is, although the fact that we don't need to update the team-specific pathfinders unless the shared one has been updated isn't necessarily obvious. It could change due to fairly minor modifications - for example, if the shared pathfinder ignored all doors (instead of treating all of them as blocking as it currently does), then this wouldn't work.

Reply:
That's an easy solve though isn't it - this method exists solely for this purpose, so you can just rename it to be more clear.
Heck, I don't even think the method really belongs in movableman, it's just there cause that's where Data put it. It'd be perfectly reasonable as a private Scene method imo -  `EraseOrRedrawDoorTerrainLayerForTeamPathfindinging` or something like that.